owner/src/main/java/org/the_chance/honeymart/ui/addCategory/AddCategoryViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package org.the_chance.honeymart.ui.addCategory
+
+import android.util.Log
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.update
+import org.the_chance.honeymart.ui.base.BaseViewModel
+import org.the_chance.honeymart.domain.usecase.AddToCategoryUseCase
+import org.the_chance.honeymart.domain.usecase.GetAllCategoriesInMarketUseCase
+import org.the_chance.honeymart.domain.util.ErrorHandler
+import javax.inject.Inject
+
+@HiltViewModel
+class AddCategoryViewModel @Inject constructor(
+    private val addToCategoryUseCase: AddToCategoryUseCase,
+    private val getAllCategories: GetAllCategoriesInMarketUseCase
+): BaseViewModel<AddCategoryUIState, AddCategoryUIEffect>(AddCategoryUIState()),
+    AddCategoryListener {
+
+    override val TAG: String = "addCategory"
+
+    init {
+        getCategoryImages()
+        getAllCategory(8)
+    }
+
+    private fun getCategoryImages(){
+        _state.update { it.copy(isLoading = false, categoryImages = categoryImages.map { it.toCategoryImageUIState() }) }
+    }
```

Comment:
mapping process should be separated within the function of onGetCategoryImagesSucces